packages/web-components/src/components/va-date-text-input/va-date-text-input.tsx

```diff
@@ -0,0 +1,279 @@
+    }
+    if (target.classList.contains('input-day')) {
+      currentDay = target.value;
+    }
+    if (target.classList.contains('input-year')) {
+      currentYear = target.value;
+    }
+
+    this.value = `${currentYear}-${currentMonth ? currentMonth : ''}-${
+      currentDay ? currentDay : ''
+    }`;
+
+    // This event should always fire to allow for validation handling
+    this.dateChange.emit(event);
+  };
+
+  private handleDateKey = (event: KeyboardEvent) => {
+    // Allow 0-9, Arrow Left, Arrow Right, Delete, Backspace and Tab to clear data or move to next field
+    const validKeys = [
+      '0',
+      '1',
+      '2',
+      '3',
+      '4',
+      '5',
+      '6',
+      '7',
+      '8',
+      '9',
+      'Backspace',
+      'ArrowRight',
+      'ArrowLeft',
+      'Tab',
+      'Delete',
+    ];
+    if (validKeys.indexOf(event.key) < 0) {
+      event.preventDefault();
+      return false;
+    }
+  };
+
+  /**
+   * Whether or not an analytics event will be fired.
+   */
+  @Prop() enableAnalytics: boolean;
+
+  /**
+   * The event used to track usage of the component. This is emitted when an
+   * input value changes and enableAnalytics is true.
+   */
+  @Event({
+    eventName: 'component-library-analytics',
+    composed: true,
+    bubbles: true,
+  })
+  componentLibraryAnalytics: EventEmitter;
+
+  connectedCallback() {
+    i18next.on('languageChanged', () => {
+      forceUpdate(this.el);
+    });
+  }
+
+  disconnectedCallback() {
+    i18next.off('languageChanged');
+  }
+
+  render() {
+    const {
+      required,
+      label,
+      name,
+      error,
+      handleDateBlur,
+      handleDateChange,
+      handleDateKey,
+      value,
+    } = this;
+
+    const [year, month, day] = (value || '').split('-').map(val => val);
+
+    // Error attribute should be leveraged for custom error messaging
+    // Fieldset has an implicit aria role of group
+    return (
+      <Host value={value} error={error} onBlur={handleDateBlur}>
+        <fieldset aria-label="Input month and day fields as two digit XX and four digit year format XXXX">
+          <legend>
+            {label}{' '}
+            {required && (
+              <span class="required">(*{i18next.t('required')})</span>
+            )}
+            <div id="dateHint">Use the following format: MM DD YYYY</div>
+          </legend>
+          {error && (
+            <span class="error-message" role="alert">
+              <span class="sr-only">Error</span> {error}
+            </span>
+          )}
+          <slot />
+          <div class="date-container">
+            <va-text-input
+              label="Month"
+              name={`${name}Month`}
+              maxlength={2}
+              minlength={2}
+              pattern="[0-9]*"
+              aria-describedby="dateHint"
+              // Value must be a string
+              // if NaN provide empty string
+              value={month ? month.toString() : ''}
```

Comment:
Would we be able to simplify this? And if the NaN is leftover from an earlier `parseInt()` call, would things still work if the value we passed in was `undefined`? If so then we might be able to get rid of the `.map()` in my previous comment.

Reply:
Do we still need the ternary?

Reply:
Good call updated it to `month?.toString()`